### ISSUE TITLE
The authorization header cannot be configured on the interceptor

### ISSUE THREAD

**Issue by eliasjtg** [flutterchina/dio#611]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |   flutter   |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |   1.12.13+hotfix.5   |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |  3.0.7    |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |  3.5.3    |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |   100%   |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

I made an implementation to use an Interceptor that added an Authorization header.

```
Dio getDio() {

  Dio dio = new Dio(new BaseOptions(
    baseUrl: getHostname().toString(),
    contentType: ContentType.json.toString(),
    responseType: ResponseType.json
  ));
  dio.interceptors.addAll([
    // Append authorization
    InterceptorsWrapper(onRequest: (RequestOptions options) {
      dio.lock();
      (new storage.FlutterSecureStorage()).read(key: 'access_token').then((token) {
        if(token != null) {
          dio.options.headers[HttpHeaders.authorizationHeader] = 'Bearer ' + token;
        }
      }).whenComplete(() {
        dio.unlock();
        return options;
      });
    }),
    // Append language
    InterceptorsWrapper(onRequest: (RequestOptions options) {
      if(! options.headers.containsValue("LANGUAGE")) {
        dio.lock();
        Language.lang.then((value) {
          options.headers["LANGUAGE"] = value;
        }).whenComplete(() {
          dio.unlock();
          return options;
        });
      }
    }),
    // Debug request
    LogInterceptor(
      requestBody: true,
      responseBody: true
    )
  ]);
  }
  return dio;
}
```

 When debug the request, my Authorization header isn't set, but the LANGUAGE header is set:

```
I/flutter ( 7485): *** Request ***
I/flutter ( 7485): uri: https://api.tribuemprende.com/oauth/logout
I/flutter ( 7485): method: GET
I/flutter ( 7485): responseType: ResponseType.json
I/flutter ( 7485): followRedirects: true
I/flutter ( 7485): connectTimeout: 0
I/flutter ( 7485): receiveTimeout: 0
I/flutter ( 7485): extra: {}
I/flutter ( 7485): headers:
I/flutter ( 7485):  content-type: application/json; charset=utf-8
I/flutter ( 7485):  LANGUAGE: es
I/flutter ( 7485): data:
I/flutter ( 7485): null
I/flutter ( 7485): 
I/flutter ( 7485): *** DioError ***:
I/flutter ( 7485): uri: https://api.tribuemprende.com/oauth/logout
I/flutter ( 7485): DioError [DioErrorType.RESPONSE]: Http status error [511]
I/flutter ( 7485): uri: https://api.tribuemprende.com/oauth/logout
I/flutter ( 7485): statusCode: 511
I/flutter ( 7485): headers:
I/flutter ( 7485):  x-nob-app: Api-request
I/flutter ( 7485):  access-control-allow-headers: FIREBASE-TOKEN, DEVICE-TYPE,Origin, Bearer-Token, Content-Type, Accept, Authorization, X-Requested-With, X-nob-app, LANGUAGE
I/flutter ( 7485):  access-control-allow-credentials: true
I/flutter ( 7485):  cache-control: no-cache, private
I/flutter ( 7485):  date: Wed, 18 Dec 2019 15:36:20 GMT
I/flutter ( 7485):  access-control-allow-origin: *
I/flutter ( 7485):  access-control-allow-methods: GET, POST, OPTIONS, PUT, DELETE
I/flutter ( 7485):  content-length: 50
I/flutter ( 7485):  content-type: application/json
I/flutter ( 7485):  server: Apache/2.4.29 (Ubuntu)
I/flutter ( 7485): Response Text:
I/flutter ( 7485): {"error":true,"message":"No Autorizado","body":{}}
```

But when add this explicity in the request:

```
Options opts = Options(
      headers: {
        HttpHeaders.authorizationHeader: 'Bearer ' + await (new storage.FlutterSecureStorage()).read(key: 'access_token')
      }
  );
  
getDio().get('oauth/logout', options: opts);
```

```
I/flutter ( 7485): *** Request ***
I/flutter ( 7485): uri: https://api.tribuemprende.com/oauth/logout
I/flutter ( 7485): method: GET
I/flutter ( 7485): responseType: ResponseType.json
I/flutter ( 7485): followRedirects: true
I/flutter ( 7485): connectTimeout: 0
I/flutter ( 7485): receiveTimeout: 0
I/flutter ( 7485): extra: {}
I/flutter ( 7485): headers:
I/flutter ( 7485):  content-type: application/json; charset=utf-8
I/flutter ( 7485):  authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImp0aSI6IjY0NmUxZTA3MjU4YmNjMTdjZGMxOTliZDBjYzIzNTRmYjcwYTkzYjkzNzk0MzVjYWY2NGY5NDU3YWM0OGUyMmI4NjY1MmEwMjRjMzE1M2M5In0.eyJhdWQiOiIxIiwianRpIjoiNjQ2ZTFlMDcyNThiY2MxN2NkYzE5OWJkMGNjMjM1NGZiNzBhOTNiOTM3OTQzNWNhZjY0Zjk0NTdhYzQ4ZTIyYjg2NjUyYTAyNGMzMTUzYzkiLCJpYXQiOjE1NzY2ODMzNTQsIm5iZiI6MTU3NjY4MzM1NCwiZXhwIjoxNjA4MzA1NzU0LCJzdWIiOiIxNTY1NzM0MTkyNjk2MTQ2MDAwIiwic2NvcGVzIjpbXX0.OGrxfPyGlbvLAIWhS-_VbG-DYLY8zE0G2diA5_lq9kG0jzpu9TwpM5eF4A1QuVT4SmwUlw5j9DXr4456_mtKiXjH7loXkIMtouEePlFZbuDRs9gkMvQz-98VxlVOC26PlYg8UEAhKWcHh6m_MgDg4PJJ7373-c_w3IGAudfvao_-yTrwyb5Tc3Hn8633yuyFuVQ5EFXlFWvmJ1j7Mn4aSfXA3GTiOBqy9sEkQg_IYm5kfW-dh946cnyxR3IBCneXtRsf7sHtoEyMB7hVbZoBcSrTk_dI4TOcQTOuML2AWHjKEzQp3WAbpWhd7QktDT_UcMGuLdKoST3wli8EO5gRyUx7cRMW2euqxPLFgCP_m9nOg62AT-3m-clQm5QQPtZHmkynSvd0Ks6T_LgkN1re2qpDSZDfu2JKN0djtmjP042xJ2cBp5pr03UYnoF1HrWhCk1giWzJHjgz0w4sETI6vtSk4uV4c1rZ80PvaxNLqoASMJ8jQSWJynuaDanKUoh3q87Q89xZBxo-TO87DRi5qNStgPKVxIhuXEnkD21dOMx6BhckWYxTNzZem7-6_k0703myZVTTaS0M
I/flutter ( 7485):  LANGUAGE: es
I/flutter ( 7485): data:
I/flutter ( 7485): null
I/flutter ( 7485): 
I/flutter ( 7485): *** Response ***
I/flutter ( 7485): uri: https://api.tribuemprende.com/oauth/logout
I/flutter ( 7485): statusCode: 200
I/flutter ( 7485): headers:
I/flutter ( 7485):  x-nob-app: Api-request
I/flutter ( 7485):  access-control-allow-credentials: true
I/flutter ( 7485):  access-control-allow-headers: FIREBASE-TOKEN, DEVICE-TYPE,Origin, Bearer-Token, Content-Type, Accept, Authorization, X-Requested-With, X-nob-app, LANGUAGE
I/flutter ( 7485):  cache-control: no-cache, private
I/flutter ( 7485):  date: Wed, 18 Dec 2019 15:45:48 GMT
I/flutter ( 7485):  access-control-allow-origin: *
I/flutter ( 7485):  vary: Authorization
I/flutter ( 7485):  access-control-allow-methods: GET, POST, OPTIONS, PUT, DELETE
I/flutter ( 7485):  content-length: 71
I/flutter ( 7485):  content-type: application/json
I/flutter ( 7485):  server: Apache/2.4.29 (Ubuntu)
I/flutter ( 7485): Response Text:
I/flutter ( 7485): {"error":false,"message":"Sesión cerrada exitosamente!","body":{}}
I/flutter ( 7485): 
```

My environment:

```
flutter doctor -v
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.14.6 18G95, locale en-US)
    • Flutter version 1.12.13+hotfix.5 at /Users/imac/development/flutter
    • Framework revision 27321ebbad (8 days ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/imac/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.7.2

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] Connected device (1 available)
    • Moto G 5 Plus • ZY224RFXTX • android-arm • Android 8.1.0 (API 27)

• No issues found!

```

**Comment by procodingtools** [comment]
Hi, there is "trap" in  this Dio package. i also had this problem because when Dio sends headers it will lowerCasing all headers keys.
So if you have access to the server you must lowercase your key.
Eg. [PHP]
`$token = array_change_key_case($this->input->request_headers(), CASE_LOWER)['authorization'];`
else, browse the package src and try to find the method which lowerCasing the headers maps key.
I'm waiting for Dio Team to fix or explain this bug.

**Comment by eliasjtg** [comment]
It's strange, because in both situations use HttpHeaders.authorizationHeader, in interceptor not work but in request options work.

In the debug of the request does not send in header not even in lower case.

So I think it must be a different problem, but thanks for interest.

**Comment by eliasjtg** [comment]
No stale :(

**Comment by eliasjtg** [comment]
:(

**Comment by iyussupov** [comment]
Hey!
`options.headers.addAll({"authorization": "Bearer $token"});`
Works for me!

**Comment by pppshiwen** [comment]
if(token != null) {
          options.headers[HttpHeaders.authorizationHeader] = 'Bearer ' + token;
        }

Works for me

**Comment by eliasjtg** [comment]
> if(token != null) {
> options.headers[HttpHeaders.authorizationHeader] = 'Bearer ' + token;
> }
> 
> Works for me

Thanks you! <3
